### PR TITLE
🥗 Swipe-able weekly food tabs

### PR DESCRIPTION
**Pull request by BuildmodeOne**
Despite the valuable comments in #198, I would suggest the design I already shown in #198 (see GIF below) to show more than 
5 days for the food tab. 

I've been trying around for a while now and feel this design is the most user friendly.

In combination with the recent changes regarding the restaurant listing, the design is not too intrusive and helps with the usability regarding the week selection.

Aufzeichnung-2023-03-08-031453

### CODE REVIEW

**Comment by alexhorn** [comment]
Honestly I'm not convinced about this UI pattern. 🙈
I would delay this until after the relaunch.

**Comment by BuildmodeOne** [comment]
New design suggested in #198

File: rogue-thi-app/pages/food.js [hunk 1]
```diff
@@ -44,15 +46,26 @@ export default function Mensa () {
...
       try {
         const days = await loadFoodEntries(selectedRestaurants)
-        setFoodDays(days.slice(0, 5))
+
+        /**
+         * new Date(days[0].timestamp).getDay()
+         * api returns full next 2 weeks on weekends, so just `new Date()`
+         * to calculate the days till friday would be wrong (higher than 5)
+         * */
+        const daysTillFriday = 5 - new Date(days[0].timestamp).getDay() + 1
```
Comment (Robert27) [inline rogue-thi-app/pages/food.js:65]: Undefined error if no data is available, e.g. during the semester break or or if no restaurants are selected. Adding the `optional chaining` operator might help.

File: rogue-thi-app/pages/food.js [hunk 2]
```diff
@@ -218,10 +237,29 @@ export default function Mensa () {
...
-          </SwipeableTabs>
+        <div className={styles.weekSelector}>
+          <Button className={styles.prevWeek} variant="link" onClick={() => setPage(0)} disabled={page === 0}>
+            <FontAwesomeIcon title="Woche zurück" icon={faChevronLeft} />
+          </Button>
+          <div className={styles.currentWeek}>
+            {page === 0 && WORD_THIS_WEEK}
+            {page === 1 && WORD_NEXT_WEEK}
+          </div>
+          <Button className={styles.nextWeek} variant="link" onClick={() => setPage(1)} disabled={page === 1}>
```
Comment (Robert27) [inline rogue-thi-app/pages/food.js:248]: If there is no data available only for next week, it is still possible to switch to the empty next week. I think we should disable this button then.

**Review by Robert27 — CHANGES_REQUESTED** [review]
Thanks for your update. I marked a few small bugs, but apart from that the code looks good.

File: rogue-thi-app/pages/food.js [hunk 3]
```diff
@@ -218,10 +237,29 @@ export default function Mensa () {
...
+          </Button>
+        </div>
+
+        <ReactPlaceholder type="text" rows={20} ready={currentFoodDays && futureFoodDays}>
+          <SwipeableViews index={page} onChangeIndex={idx => setPage(idx)}>
+            <SwipeableTabs className={styles.tab} fillers={dayFiller}>
+              {currentFoodDays && currentFoodDays.map((day, idx) => renderMealDay(day, idx))}
+            </SwipeableTabs>
+
+            <SwipeableTabs className={styles.tab}>
```
Comment (Robert27) [inline rogue-thi-app/pages/food.js:259]: If the entire next week is not available, such as with holidays, I would add the fillerdays here as well for design consistency.

**Review by Robert27 — CHANGES_REQUESTED** [review]

File: rogue-thi-app/pages/food.js (hunk 3 above)
Comment (BuildmodeOne) [reply rogue-thi-app/pages/food.js:259]: Due to the `daysTillFriday` variable being set to zero if no data is available, the fillers will be added.

**Review by M4GNV5 — APPROVED** [review]

File: rogue-thi-app/components/SwipeableTabs.js [hunk 4]
```diff
@@ -9,14 +9,21 @@ import SwipeableViews from 'react-swipeable-views'
...
-export default function SwipeableTabs ({ className, children }) {
+export default function SwipeableTabs ({ className, children, fillers }) {
   const [page, setPage] = useState(0)
 
   return (
     <div className={className}>
       <Nav variant="pills" activeKey={page.toString()} onSelect={key => setPage(parseInt(key))}>
+        {fillers && fillers.map((child, idx) =>
+          <Nav.Item key={idx} >
+            <Nav.Link className='disabled'>
```
[the remark this answers is not in the document]
Comment (Robert27) [reply rogue-thi-app/components/SwipeableTabs.js:20]: I am pretty sure it updates daily but we will see tomorrow ig

[the remark this answers is not in the document]
Comment (M4GNV5) [reply rogue-thi-app/components/SwipeableTabs.js:20]: The data seems available in the xml, but we filter it out, here: https://github.com/neuland-ingolstadt/neuland.app/blob/develop/rogue-thi-app/pages/api/mensa.js#L46-L48

**Pull request merged**